### ISSUE TITLE
Generic arrow function with multi line parameters and implied return body  indentation detection

### ISSUE THREAD

**Issue by ahumeau** [gajus/eslint-plugin-flowtype#360]
I am getting inconsistencies with how the correct indent is determined for the body of generic arrow functions which parameters are declared on multiple lines.

__Input:__
```js
/* @flow */


const VALID_genericSingleLineParams = <T>(bar: T[]) => (
  bar
)


const VALID_notGenericMultiLineParams = (
  bar
) => (
  bar
)


const INVALID_genericMultiLineParams = <T>(
  bar: T[]
) => (
  bar
)

```

__ESLint output:__
```
test.js
  19:1  error  Expected indentation of 4 spaces but found 2    indent
  20:1  error  Expected indentation of 2 spaces but found 0    indent
```

**Comment by gajus** [comment]
The error in the issue is coming from ESLint. Which is the expected behaviour?

**Comment by ahumeau** [comment]
I indeed did not know where was the source of the inconsistency but since it was related to a Flow syntax addition I assumed this was the place to open an issue, I apologize if I was mistaken.

The expected behaviour is that the expected indentation for the body of the third function definition be the same as for the others, it is currently one level higher.

**Comment by gajus** [comment]
> The expected behaviour is that the expected indentation for the body of the third function definition be the same as for the others, it is currently one level higher.

This is ESLint bug by the looks of it.

**Comment by gajus** [comment]
Open an issue with ESLint and reference this thread please.
